src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -1091,6 +1108,21 @@ protected String getBuildName(final Run<?, ?> run) {
         }
     }
 
+    /**
+     * Returns the build url to be pushed. This will select the specifically overwritten build url
+     * or get the build url from the {@link DisplayURLProvider}.
+     *
+     * @param run the run to notify Bitbucket of
+     * @return the url of the run
+     */
+    protected String getBuildUrl(final Run<?, ?> run) {
+        if (buildUrl != null && buildUrl.trim().length() > 0) {
```

Comment:
You can test for empty directly:
```suggestion
        if (buildUrl != null && !buildUrl.trim().isEmpty()) {
```